04_health/exposure-to-trauma/exposure-to-trauma-2020.Rmd

```diff
@@ -0,0 +1,434 @@
+    - *U02 (Sequelae of terrorism)
+    - *U03 (Terrorism Intentional (Suicide))
+    - V01-V99 (Transport accidents)
+    - W00-X59 (Other external causes of accidental injury)
+    - X60-X84 (Intentional self-harm)
+    - X85-Y09 (Assault)
+    - Y10-Y34 (Event of undetermined intent)
+    - Y35-Y36 (Legal intervention and operations of war)
+    - Y85 (Sequelae of transport accidents)
+    - Y86 (Sequelae of other accidents)
+    - Y87 (Sequelae of intentional self-harm, assault and events of undetermined intent)
+    - Y89 (Sequelae of other external causes)
+7. Other options:
+  + Show Totals
+  + Show Zero Values
+  + Show Suppressed Values
+  + Precision: 1 decimal places
+  + Data Access Timeout: 10 minutes
+  
+The specifications for the 2020 extract are saved [here](https://wonder.cdc.gov/controller/saved/D76/D377F651). Note that counties with fewer than ten deaths (for all the ICD-10 codes specified) are suppressed. Population figures are also suppressed when the population represents fewer than ten persons. Mortality rates are marked as "unreliable" when the death count is less than 20. Rates are marked as "not applicable" when the population denominator figure is unavailable, such as persons of "not stated" or unknown age or Hispanic origin.  
+
+Once the extract is created, select "Export" to generate a .txt file. Move the .txt file into the appropriate directory to read in the data.
+
+```{r}
+# List all the files in the `bridged-race` folder
+paths <- list.files(here::here("04_health", "exposure-to-trauma", "data", "bridged-race"),
+                    pattern = "[.]txt$", full.names = TRUE)
+
+# Read the files in and append them
+deaths_14_20 <- paths |> 
+  set_names(basename) |> 
+  map(utils::read.delim) |> 
+  list_rbind(names_to = "year") |> 
+  mutate(year = parse_number(year))
+
+# Rename variables and remove `Notes` column
+deaths_14_20 <- deaths_14_20 |>
+  select(
+    year,
+    county_name = County,
+    GEOID = County.Code,
+    deaths = Deaths,
+    population = Population,
+    # We are only interested in the age-adjusted rates, so ignoring crude rates
+    rate_injury_death_crude = Crude.Rate,
+    # rate_injury_death_lb = Crude.Rate.Lower.95..Confidence.Interval,
+    # rate_injury_death_ub = Crude.Rate.Upper.95..Confidence.Interval,
+    # rate_injury_death_se = Crude.Rate.Standard.Error,
+    rate_injury_death = Age.Adjusted.Rate,
+    rate_injury_death_lb = Age.Adjusted.Rate.Lower.95..Confidence.Interval,
+    rate_injury_death_ub = Age.Adjusted.Rate.Upper.95..Confidence.Interval,
+    rate_injury_death_se = Age.Adjusted.Rate.Standard.Error
+  )
+
+# Remove empty rows resulting from deleting the `Notes` column, and remove the total row
+deaths_14_20 <- deaths_14_20 %>%
+  filter(!is.na(GEOID))
+
+# Change all variables to numeric and replace suppressed or unreliable with missing
+deaths_14_20 <- deaths_14_20 %>%
+  # mutate_all(~ifelse(. %in% c("Suppressed", "Unreliable"), NA, .)) %>%
```

Comment:
I agree the commented out approach is better. I initially went with the currently un-commented out approach because it replaced those text values with `NA` AND converted the rates and counts from character variables to numeric. With the commented out approach, I had to convert the variables in a separate step, so I initially chose not to use that approach. However, I agree it is clearer and have gone with the commented out approach instead.